src/supermarket/app/controllers/cookbooks_controller.rb

```diff
@@ -308,4 +309,17 @@ def apply_filters
       @cookbooks = @cookbooks.filter_badges(params[:badges])
     end
   end
+
+  def classify_metrics(metric_results)
```

Comment:
More thinking in the fresh light of morning and after a cup of coffee ...

I cannot think of a current case of asking for a cookbook_version's metric_results in which we would not also want the associated quality_metric. If that is true, it might be worth baking the `includes()` into the relationship:

```diff
class CookbookVersion < ActiveRecord::Base
...
-  has_many :metric_results
+  has_many :metric_results, -> { includes :quality_metric }
```

I've been playing around with scopes further to push this query logic into named things on the models.